src/components/form/Form.js

```diff
@@ -0,0 +1,215 @@
+    props.onClose();
+    setTimeout(clearValues, 1000);
+    setTimeout(() => {
+      // Display form instead of confirmation message when reopening popup
+      setDisplayForm(true);
+      setIsCompleted(false);
+    }, 1000);
+    window.removeEventListener('keydown', handleEscClose);
+  }
+
+  function handleModalClick(e) {
+    if (e.target.classList.contains('modal_open')) {
+      closeModal();
+    }
+  }
+
+  function handleEscClose(e) {
+    if (e.key === 'Escape') {
+      closeModal();
+    }
+  }
+  window.addEventListener('keydown', handleEscClose);
+
+  // When user clicks on submit button, form will be validated
+  function handleSubmit(e) {
+    e.preventDefault();
+    setInputErrors(validate(inputValues));
+    setOnSubmitClick(true);
+  }
+
+  // Form validation handler
+  const validate = (values) => {
+    const errors = {};
+    const regex = /^[^\s@]+@[^\s@]+\.[^\s@]{2,}$/i;
+
+    if (!values.email) {
+      errors.email = 'Email address is required';
+    } else if (!regex.test(values.email)) {
+      errors.email = 'Invalid email format';
+    }
+
+    if (!values.company) {
+      errors.company = 'Company name is required';
+    }
+
+    if (!values.description) {
+      errors.description = 'Task description is required';
+    }
+
+    if (!values.name) {
+      errors.name = 'Name is required';
+    }
+
+    return errors;
+  };
+
+  // If there are no errors the form will be submitted
+  React.useEffect(() => {
+    if (Object.keys(inputErrors).length === 0 && onSubmitClick) {
+      submit();
+    }
+  }, [inputErrors]);
+
+  // Close form and display confirmation message
+  function submit() {
+    setDisplayForm(false);
+    setIsCompleted(true);
+  }
+
+  return (
+    <div className={`modal ${props.isOpen ? 'modal_open' : ''}`} onClick={handleModalClick}>
+      <div className="modal__content" style={{ display: displayForm ? 'block' : 'none' }}>
+        <button
+          className="modal__close-button modal__close-button_form"
+          type="button"
+          onClick={closeModal}
+        />
+        <h2 className="modal__title">Delegate us your task! Fill in this form:</h2>
+        <form className="form" onSubmit={handleSubmit} noValidate>
+          <label className="form__label form__label_type_company" htmlFor="company-name">
+            Your company name *
+            <input
+              type="text"
+              id="company"
+              name="company"
+              className="form__input form__input_type_company"
+              autoComplete="off"
+              placeholder="Company Name"
+              onChange={handleChange}
+              value={inputValues.company}
+            />
+            <span className={`form__error ${inputErrors.company && 'form__error_visible'}`}>
+              {inputErrors.company}
+            </span>
+          </label>
+
+          <label className="form__label form__label_type_email" htmlFor="email">
+            Your email *
+            <input
```

Comment:
do you need a "required" attribute here?

Reply:
Good point, just added them!